src/workflow/unlock.c

```diff
@@ -122,37 +143,143 @@ keystore_error_t workflow_unlock_and_handle_error(const char* password)
     return unlock_result;
 }
 
-bool workflow_unlock(void)
+static void _unlock_cleanup_password(unlock_data_t* data)
+{
+    if (data->password) {
+        util_zero(data->password, sizeof(data->password));
+        free(data->password);
+        data->password = NULL;
+    }
+}
+
+/**
+ * Callback invoked when the password entry completes.
+ */
+static void _unlock_password_completed(const char* password, void* param)
```

Comment:
not about this particular callback, but in general: before i went through a lot of effort to reduce "callback hell", and now it seems we are going callback heavy again.

it might be required, but it would be nice if we could figure out something that is easier to read/follow than lots of ad-hoc state machines and callbacks. 

before, the callbacks were hard to maintain and bug-prone to a point where we did not introduce new workflows because of it.

cc @NickeZ

Reply:
Discussion ongoing offline

Reply:
As discussed in other places, now that we have proper asynchronous workflows we can't have code that look synchronous/sequential since C doesn't have support for async.

I'm totally fine with this PR.